pkg/gitops/update_repo.go

```diff
@@ -46,24 +44,14 @@ func (cr *CustomRepo) modifyFile(event auditv1.Event) error {
 	if err != nil {
 		return fmt.Errorf("unable to marshal new resource config: %w", err)
 	}
-	path := getAbsRepoPath(cr.Dir, event)
-	if cr.StorageMode == StorageModeDisk {
-		if _, err := os.Stat(path); os.IsNotExist(err) {
-			os.Mkdir(path, 0700)
-		}
-		path += getFileName(event)
-		if err := ioutil.WriteFile(path, y, 0600); err != nil {
-			return fmt.Errorf("unable to write/update file in repository: %w", err)
-		}
-	} else {
-		path += getFileName(event)
-		newfile, err := cr.Fs.Create(path)
-		if err != nil {
-			return fmt.Errorf("unable to create file at: %s: %w", path, err)
-		}
-		newfile.Write(y)
-		newfile.Close()
+	path := getAbsRepoPath("", event)
+	path += getFileName(event)
+	newfile, err := cr.Fs.Create(path)
+	if err != nil {
+		return fmt.Errorf("unable to create file at: %s: %w", path, err)
 	}
+	newfile.Write(y)
+	newfile.Close()
```

Comment:
use `defer newfile.Close()` just before `newfile.Write(y)`. It's a more common pattern and makes it easy to verify that you never forget to close a file.